### PR TITLE
Add NeighborPackagedData to GHMHD

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_NeighborPackagedData.cpp [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+      neighbor_data, 1.0, std::move(element_map), moving_mesh_map.get_clone(),
+      dummy_volume_mesh_velocity, normal_vectors, time,
+      clone_unique_ptrs(functions_of_time),
+      // Note: These damping functions all assume Grid==Inertial. We need to
+      // rescale the widths in the Grid frame for binaries.
+      std::unique_ptr<DampingFunction>(  // Gamma0, taken from SpEC BNS
+          std::make_unique<GeneralizedHarmonic::ConstraintDamping::
+                               GaussianPlusConstant<3, Frame::Grid>>(
+              0.01, 0.09 * 1.0 / 1.4, 5.5 * 1.4, std::array{0.0, 0.0, 0.0})),
+      // Gamma1, taken from SpEC BNS
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_NeighborPackagedData.cpp]: move these comments up to where `gamma_1` and `gamma2` are defined

File: tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_NeighborPackagedData.cpp [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+  return max_rel_error;
+}
+
+// [[Timeout, 10]]
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GhValenciaDivClean.Subcell.NeighborPackagedData",
+    "[Unit][Evolution]") {
+  // This tests sets up a cube [-4,4]^3 in a TOV star spacetime and verifies
+  // that the time derivative vanishes. Or, more specifically, that the time
+  // derivative decreases with increasing resolution and is below 1.0e-6.
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_NeighborPackagedData.cpp]: looks like 1.0e-4 ...

**Review by kidder — CHANGES_REQUESTED** [review]
squash

**Comment by nilsdeppe** [comment]
Done and rebased. Thanks for the review :D

**Review by kidder — APPROVED** [review]

**Pull request merged**